core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```
@@ -87,12 +95,50 @@
+                throw new RuntimeException("Failed to populate MSF4J Configuration. Config Provider is Null.");
+            }
+            //Standalone mode
+            String deploymentYamlPath = System.getProperty(MSF4JConstants.DEPLOYMENT_YAML_SYS_PROPERTY);
+            if (deploymentYamlPath == null || deploymentYamlPath.isEmpty()) {
+                log.info("System property '" + MSF4JConstants.DEPLOYMENT_YAML_SYS_PROPERTY +
+                         "' is not set. Default deployment.yaml file will be used.");
+                URL deploymentYamlUrl = MSF4JMessageProcessor.class.getResource("/deployment.yaml");
+                try {
+                    FileUtils.copyURLToFile(deploymentYamlUrl,
+                                            Paths.get(MSF4JConstants.DEPLOYMENT_YAML_FILE_NAME).toFile());
+                    deploymentYamlPath = Paths.get(MSF4JConstants.DEPLOYMENT_YAML_FILE_NAME).toString();
+                } catch (IOException e) {
+                    throw new RuntimeException(e.getMessage());
+                }
+            } else if (!Files.exists(Paths.get(deploymentYamlPath))) {
+                String msg = "Couldn't find " + deploymentYamlPath;
+                throw new RuntimeException(msg);
+            }
+
+            try {
+                configProvider = ConfigProviderFactory.getConfigProvider(Paths.get(deploymentYamlPath), null);
+                DataHolder.getInstance().setConfigProvider(configProvider);
+            } catch (ConfigurationException e) {
+                throw new RuntimeException("Error loading deployment.yaml Configuration", e);
+            }
+        }
+
+        try {
+            msf4JConfig = DataHolder.getInstance().getConfigProvider().getConfigurationObject(MSF4JConfig.class);
+        } catch (ConfigurationException e) {
+            log.error("Error loading configurations from deployment.yaml", e);
+            msf4JConfig = new MSF4JConfig();
```

New instantiation can again cause an exception. Better to throw a runtime exception as in the previous case.